logstash-core/lib/logstash/environment.rb

```diff
@@ -40,8 +40,18 @@ module Environment
             Setting::String.new("http.host", "127.0.0.1"),
             Setting::PortRange.new("http.port", 9600..9700),
             Setting::String.new("http.environment", "production"),
+            Setting::String.new("queue.type", "persisted", true, ["persisted", "memory", "synchronous"]),
+            Setting::Numeric.new("queue.page_capacity", 250 * 1024 * 1024),
+            Setting::Numeric.new("queue.queue_capacity", 250 * 1024 * 1024)
+
   ].each {|setting| SETTINGS.register(setting) }
 
+  queue_file_path = ::File.join(SETTINGS.get("path.data"), "queue")
+  # Here we have to create the directory even though it might be overridden by settings.yml
+  # We should consider a Setting::LazyDirectory class that creates the directory on first 'get'
+  FileUtils.mkdir_p([queue_file_path])
+  SETTINGS.register Setting::WritableDirectory.new("path.queue", queue_file_path)
```

Comment:
I did this temporary hack [in the rake test task](https://github.com/elastic/logstash/blob/7f66d8bd176d01991135d330935ce59bd1ae0469/rakelib/test.rake#L10-L13) - I guess we should use the same mecanism there ... but is SETTINGS exposed at this point? seems tricky